src/main/java/duke/execution/Parser.java

```diff
@@ -0,0 +1,150 @@
+            } else {
+                ordinalIndicator = "th";
+            }
+
+            String outputDate = intDay + ordinalIndicator + " of " + month + " " + year + ", " + time;
+            return outputDate;
+        } catch (ParseException e) {
+            System.out.println("isDate has bugs");
+            System.out.println(e.getMessage());
+            return "";
+        }
+    }
+
+    private static Command parseForCommand(String action, String txt) throws DukeException {
+        switch (action) {
+        case "bye":
+            return new ByeCommand(txt);
+
+        case "list":
+            return new ListCommand(txt);
+
+        case "done":
+            if (txt.isEmpty()) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! The description of a done cannot be empty.");
+            }
+            return new DoneCommand(txt);
+
+        case "todo":
+            if (txt.isEmpty()) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! The description of a todo cannot be empty.");
+            }
+            return new TodoCommand(txt);
+
+        case "deadline":
+            if (txt.isEmpty()) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! The description of a deadline cannot be empty.");
+            } else if (!txt.contains("/by")) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! Deadline input should include '/by'.");
+            }
+            return new DeadlineCommand(txt);
+
+        case "event":
+            if (txt.isEmpty()) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! The description of an event cannot be empty.");
+            } else if (!txt.contains("/at")) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! Event input should include '/at'.");
+            }
+            return new EventCommand(txt);
+
+        case "delete":
+            if (txt.isEmpty()) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! The description of an delete cannot be empty.");
+            }
+            return new DeleteCommand(txt);
+
+        case "find":
+            if (txt.isEmpty()) {
+                System.out.println("ERROR_LOG: txt: " + txt);
+                throw new DukeException(" ☹ OOPS!!! There is no input for the search!");
+            }
+            return new FindCommand(txt);
+
+        default:
+            System.out.println("ERROR_LOG: txt: " + txt);
+            throw new DukeException(" ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private static String getRestOfInput(String[] split) {
```

Comment:
There is already an existing similar overloaded method eg. string.split("-", integer) that lets you limit the number of splits. 
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#split(java.lang.String,int)